tests/src/test/java/org/eclipse/hono/tests/client/HonoKafkaConsumerIT.java

```diff
@@ -274,13 +281,19 @@ public void testConsumerReadsLatestRecordsPublishedAfterOutOfRangeOffsetReset(fi
 
         kafkaConsumer = new HonoKafkaConsumer<>(vertx, topics, recordHandler, consumerConfig);
         // first start of consumer, letting it commit offsets
-        kafkaConsumer.start().onComplete(ctx.succeeding(v -> {
-            LOG.trace("consumer started, publish first round of records to be received by the consumer (so that it has offsets to commit)");
-            publishRecords(numTestRecordsPerTopicPerRound, "round1_", topics);
-        }));
-
-        assertThat(firstConsumerInstanceStartedAndStopped.awaitCompletion(IntegrationTestSupport.getTestSetupTimeout(),
-                TimeUnit.SECONDS)).isTrue();
+        final Promise<Void> readyTracker = Promise.promise();
+        kafkaConsumer.addOnKafkaConsumerReadyHandler(readyTracker);
```

Comment:
Such a `readyTracker` should still be added in line 323 (bottom of `testConsumerReadsLatestRecordsPublishedAfterOutOfRangeOffsetReset` test).